node/node.go

```diff
@@ -241,27 +241,22 @@ func NewNode(config *cfg.Config,
 	csMetrics, p2pMetrics, memplMetrics := metricsProvider()
 
 	// Make MempoolReactor
-	maxDataBytes := types.MaxDataBytesUnknownEvidence(
-		state.ConsensusParams.BlockSize.MaxBytes,
-		state.Validators.Size(),
-	)
-	maxGas := state.ConsensusParams.BlockSize.MaxGas
 	mempool := mempl.NewMempool(
 		config.Mempool,
 		proxyApp.Mempool(),
 		state.LastBlockHeight,
 		mempl.WithMetrics(memplMetrics),
-		mempl.WithPreCheckFilter(func(tx types.Tx) bool {
-			// We have to account for the amino overhead in the tx size as well
-			aminoOverhead := amino.UvarintSize(uint64(len(tx)))
-			return (len(tx) + aminoOverhead) <= maxDataBytes
-		}),
-		mempl.WithPostCheckFilter(func(tx types.Tx, res *abci.ResponseCheckTx) bool {
-			if maxGas == -1 {
-				return true
-			}
-			return res.GasWanted <= maxGas
-		}),
+		mempl.WithPreCheckFilter(
+			mempool.PreCheckAminoMaxBytes(
+				types.MaxDataBytesUnknownEvidence(
```

Comment:
I think this change makes it harder to read than having maxDataBytes defined above

Reply:
Is this a comment regarding your personal taste or is there strong evidence how one is superior to the other?

Reply:
Personal preference, though I believe its founded on strong evidence. 3 function calls is less clear than two function calls on an explicitly defined variable name.

Reply:
The hierarchical nesting can be followed step-by-step to understand what is invoced, where with a variable one has to backtrack. Addtionally the scope of the NewNode function is already quite large, so avoiding adding more variables is desriable. Either case should be fine, if you have strong aversion towards the current formatting we can change it.

Reply:
nah no strong opinion :)